BiometricsFingerprint.cpp

```diff
@@ -599,10 +609,13 @@ void * BiometricsFingerprint::worker_thread(void *args){
                         thisPtr->mClientCallback->onAuthenticated(devId, fid, gid, hidl_vec<uint8_t>());
                     }
                 } else {
-                    setState(sdev, STATE_IDLE);
-                    thisPtr->mClientCallback->onError(devId, FingerprintError::ERROR_CANCELED, 0);
-                    raise(SIGKILL);
-                    break;
+                      /*
+                       * Reinitialize the TZ app and parameters
+                       * to clear the TZ error generated by flooding it
+                       */
+                      fpc_close(&sdev->fpc);
```

Comment:
I can't say i really trust this to properly reinitialize, but I can't figure out what the right way to fail a HIDL hal is - there doesn't seem to be a way to kill the thread without doing nasty stuff..

Reply:
Theoretically, from what I've understood, this HAL shall never fail... so... well, that one is actually closing the tzfingerprint app, so that when we open it again it's almost like we're killing the HAL and reinitializing it, minus error reporting to Android, which messes it up